xarray/core/dataarray.py

```diff
@@ -1789,7 +1817,11 @@ def _reindex_callback(
             exclude_dims,
             exclude_vars,
         )
-        return self._from_temp_dataset(reindexed)
+
+        da = self._from_temp_dataset(reindexed)
+        da.encoding = self.encoding
```

Comment:
is this fixing something else? a test would not be a bad idea :)

Reply:
It is not fixing anything specific here, it has been moved from `Aligner` into the Dataset / DataArray callbacks when adding alignment support for Coordinates. I guess it is already tested somewhere?